src/gtest/test_sidechain_to_mempool.cpp

```diff
@@ -595,82 +752,98 @@ TEST_F(SidechainsInMempoolTestSuite, CertInMempool_QualityOfCerts) {
-    txCreationUtils::storeSidechain(*pcoinsTip, scId, sidechain, dummyCeasingMap);
-    txCreationUtils::storeCertDataHash(*pcoinsTip, scId, 0);
-    txCreationUtils::storeCertDataHash(*pcoinsTip, scId, sidechain.prevBlockTopQualityCertReferencedEpoch-1);
-
-    CAmount cswTxCoins = sidechain.balance/2;
+    CNakedCCoinsViewCache sidechainsView(pcoinsTip);
+
+    // setup sidechain initial state
+    CSidechain initialScState;
+    uint256 scId = uint256S("aaaa");
+    initialScState.creationBlockHeight = 1492;
+    initialScState.creationData.withdrawalEpochLength = 14;
+    initialScState.creationData.wCeasedVk = libzendoomc::ScVk(ParseHex(SAMPLE_VK));
+    initialScState.balance = CAmount{1000};
+    int heightWhereCeased = initialScState.GetScheduledCeasingHeight();
+
+    storeSidechainWithCurrentHeight(sidechainsView, scId, initialScState, heightWhereCeased);
+    ASSERT_TRUE(sidechainsView.GetSidechainState(scId) == CSidechain::State::CEASED);
+
+    CAmount cswTxCoins = initialScState.balance/2;
     CTxCeasedSidechainWithdrawalInput cswInput = GenerateCSWInput(scId, "aabb", cswTxCoins);
     CTransaction cswTx = GenerateCSWTx(cswInput);
 
-    CValidationState dummyState;
     libzendoomc::CScProofVerifier verifier = libzendoomc::CScProofVerifier::Disabled();
-    EXPECT_TRUE(pcoinsTip->IsTxCswApplicableToState(cswTx, dummyState, verifier));
+    EXPECT_TRUE(sidechainsView.IsScTxApplicableToState(cswTx, verifier));
 }
 
 TEST_F(SidechainsInMempoolTestSuite, ExcessiveAmountOfCSWsToCeasedSidechainIsRejected) {
-    uint256 scId = uint256S("aaa");
-    CSidechain sidechain;
-    sidechain.balance = CAmount(100);
-    sidechain.creationData.wCeasedVk = libzendoomc::ScVk(ParseHex(SAMPLE_VK));
-    sidechain.currentState = (uint8_t)CSidechain::State::CEASED;
+    CNakedCCoinsViewCache sidechainsView(pcoinsTip);
+
+    // setup sidechain initial state
+    CSidechain initialScState;
+    uint256 scId = uint256S("aaaa");
+    initialScState.creationBlockHeight = 1492;
+    initialScState.creationData.withdrawalEpochLength = 14;
+    initialScState.creationData.wCeasedVk = libzendoomc::ScVk(ParseHex(SAMPLE_VK));
+    initialScState.balance = CAmount{1000};
+    int heightWhereCeased = initialScState.GetScheduledCeasingHeight();
 
-    CSidechainEventsMap dummyCeasingMap;
-    txCreationUtils::storeSidechain(*pcoinsTip,scId, sidechain, dummyCeasingMap);
-    txCreationUtils::storeCertDataHash(*pcoinsTip, scId, 0);
+    storeSidechainWithCurrentHeight(sidechainsView, scId, initialScState, heightWhereCeased);
+    ASSERT_TRUE(sidechainsView.GetSidechainState(scId) == CSidechain::State::CEASED);
 
-    CAmount cswTxCoins = sidechain.balance*2;
+    CAmount cswTxCoins = initialScState.balance*2;
     CTxCeasedSidechainWithdrawalInput cswInput = GenerateCSWInput(scId, "aabb", cswTxCoins);
     CTransaction cswTx = GenerateCSWTx(cswInput);
 
-    CValidationState dummyState;
+    int height = pcoinsTip->GetHeight();
```

Comment:
1) `height` is not used.
2) Probably, `CSWs...` tests can be moved to `...sidechain.cpp` test file.